src/mem/superslab.h

```diff
@@ -73,8 +74,10 @@ namespace snmalloc
 
     static bool is_short_sizeclass(sizeclass_t sizeclass)
```

Comment:
It might be good to have some overview of the whole "short slab" scheme, say, at the top of `superslab.h`.  Just spitballing some text:

```
Superslabs are, to first approximation, a `CHUNK_SIZE`-sized and -aligned
region of address space, internally composed of a header (a `Superslab`
structure) followed by an array of `Slab`s, each `SLAB_SIZE`-sized and
-aligned.  Each active `Slab` holds an array of identically sized allocations
strung on an invasive free list, which is lazily constructed from a
bump-pointer allocator (see `Metaslab::alloc_new_list`).

In order to minimize overheads, Slab metadata is held externally, in `Metaslab`
structures; all `Metaslab`s for the Slabs within a Superslab are densely packed
within the `Superslab` structure itself.  Moreover, as the `Superslab`
structure is typically much smaller than `SLAB_SIZE`, a "short Slab" is
overlaid with the `Superslab`.  This short Slab can hold only allocations that
are smaller than the `SLAB_SIZE - sizeof(Superslab)` bytes; see
`Superslab::is_short_sizeclass`.  The Metaslab state for a short slabs is
constructed in a way that avoids branches on fast paths; effectively, the
object slots that overlay the `Superslab` at the start are omitted from
consideration.
```